zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -240,6 +250,16 @@ public boolean isEmpty() {
     return isEmpty.get();
   }
 
+  private void assertForeignKeyExists(final ZeebeTransaction transaction, final Object key)
+      throws Exception {
+    if (!consistencyChecksSettings.enableForeignKeys()) {
```

Comment:
:speech_balloon: Happy to duplicate the feature flag as a single switch. But I still prefer to also and most importantly have it in `ForeignKeyChecker`

Reply:
:+1: I've moved the switch to `ForeignKeyChecker`